### ISSUE TITLE
FinalLocalVariable: Wrong variable reported because of name shadowing

### ISSUE THREAD

**Issue by eekboom** [checkstyle/checkstyle#1142]
http://checkstyle.sourceforge.net/config_coding.html#FinalLocalVariable

Checkstyle reports that the "text" parameter in the "bar" method can be final, which is wrong.
It should report the "text" param in the "foo" method instead.
```Java
class Foo {
    public void foo(String text) {
        System.out.println(text);

        class Bar {
            void bar(String text) {
                text = "xxx";
            }
        }
    }
}
```

**Comment by romani** [comment]
please provide configuration and what version of checkstyle you are using.
Please try to reproduce problem with our CLI - http://checkstyle.sourceforge.net/cmdline.html#Download_and_Run

**Comment by eekboom** [comment]
Version 6.6
This is the error message:
D:\stephen\sniff\src\com\acme\MissingFinals.java:8:29: Variable 'text' sollte als final deklariert werden.
(Translation: Variable 'text' should be declared final)
Note the wrong line number '8'!
Here's the config:
```XML
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="fileExtensions" value="java"/>

    <module name="TreeWalker">
        <module name="FinalLocalVariable">
            <property name="tokens" value="VARIABLE_DEF,PARAMETER_DEF"/>
            <property name="validateEnhancedForLoopVariable" value="true"/>
        </module>
    </module>
</module>
```

**Comment by romani** [comment]
On 6.7 I had no problem with line number but issue in confirmed. Output from 6.7:
```
/var/tmp/Var.java:6:29: Variable 'text' should be declared final.
Audit done.
Checkstyle ends with 1 errors.

```

Expected: same violation but for variable at line 2.

**Comment by romani** [comment]
merged.
